pyomo/contrib/doe/fim_doe.py

```diff
@@ -729,18 +821,17 @@ def __finite_calculation(self, output_record, scena_gen):
             list_jac = []
             for i in range(len(output_record[s1])):
                 if self.scale_nominal_param_value:
```

Comment:
Add comments to explain this. Why the if statement? Mention the finite difference formula we are using.